components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -38,12 +42,48 @@ trait Validator {
 
 //TODO: These validators should be moved to separated module
 
+case class SpelExpressionParameterValidator(
+    validationExpression: String,
+    validationFailedMessage: Option[String]
+) extends ParameterValidator {
+
+  private val parser: ExpressionParser   = new SpelExpressionParser()
+  private val parsedValidationExpression = parser.parseExpression(validationExpression)
+
+  override def isValid(paramName: String, value: Any, label: Option[String])(
+      implicit nodeId: NodeId
+  ): Validated[PartSubGraphCompilationError, Unit] = {
+    val context: StandardEvaluationContext = new StandardEvaluationContext()
+    context.setVariable(paramName, value)
+    Try(parsedValidationExpression.getValue(context, classOf[Boolean])).fold(
+      e =>
+        invalid(
+          CustomParameterValidationError(
+            s"Expression evaluation failed: ${e.getMessage}",
```

Comment:
The user will see this message in the context when he/she uses the fragment. I think it won't be clear, which expression is evaluated with failure. Do you know how to change this message to make this more obvious?

Reply:
It should be more obvious now :)

Reply:
fixed